### ISSUE TITLE
Crashing app when server not found

### ISSUE THREAD

**Issue by JonnyBGod** [ShaunBaker/PredictionIO-Node-SDK#4]
```
/Users/joaoribeiro/Documents/Projects/imoglobe/node_modules/predictionio/lib/items.js:26
        deferred.resolve(res.body)
                            ^
TypeError: Cannot read property 'body' of undefined
    at /Users/joaoribeiro/Documents/Projects/imoglobe/node_modules/predictionio/lib/items.js:26:29
    at Request.callback (/Users/joaoribeiro/Documents/Projects/imoglobe/node_modules/predictionio/node_modules/superagent/lib/node/index.js:628:30)
    at ClientRequest.<anonymous> (/Users/joaoribeiro/Documents/Projects/imoglobe/node_modules/predictionio/node_modules/superagent/lib/node/index.js:596:10)
    at ClientRequest.EventEmitter.emit (events.js:95:17)
    at Socket.socketErrorListener (http.js:1547:9)
    at Socket.EventEmitter.emit (events.js:95:17)
    at net.js:441:14
    at process._tickCallback (node.js:415:13)
[nodemon] app crashed - waiting for file changes before starting...
```

**Comment by JonnyBGod** [comment]
I believe the expected behavior based on the code below would be to "deferred.reject(err)"

```
request
      .post(this._config.baseUrl + '/items.json')
      .send(params)
      .end(function (err, res) {

        if (err || !res) deferred.reject(err)
        deferred.resolve(res.body)

      })
```

however, it is getting to deferred.resolve(res.body).

**Comment by ShaunBaker** [comment]
Hey @JonnyBGod 

I have added in an extra check for the body of the response, so should be all good now.

Thanks
